Dockerfile

```diff
@@ -62,5 +62,12 @@ COPY --from=frontend /headlamp/plugins /headlamp/plugins
 # renamed it as "headlamp-server".
 RUN cd /headlamp && ln -s ./headlamp-server ./server
 
+RUN addgroup -S headlamp && adduser -S headlamp -G headlamp
+RUN chown -R headlamp:headlamp /headlamp
```

Comment:
If the user mounts the plugins in a directory other than `/headlamp` will this affect?

Reply:
Good point. Not sure, I will look into it.

Reply:
The plugins are [documented to be copied](https://headlamp.dev/docs/latest/development/plugins/building/#shipping-and-deploying-plugins) into `/headlamp/plugins` from a .plugins folder. So this does not affect the plugins copied in. Since headlamp/plugins is inside /headlamp and they have the headlamp:headlamp user.

I don't think this is a problem... can you please explain it some more?

Reply:
Yeah the documented/recommended way is to copy the plugins to `/headlamp/plugins` but the user can create a volumeMount to a directory other than `/headlamp/plugins` and then point to that plugins directory using the `--plugins-dir` flag. For example in [this](https://headlamp.dev/blog/2022/10/get-up-to-speed-deploying-headlamp-with-plugins/) blog post the plugins-dir is `/build/plugins`. In that case will the headlamp user/group need any other rights to access the plugin files.

Reply:
Understood, thanks.

Reply:
Clarifying some things, once we create `headlamp` user in the Dockerfile, we can use the same user an group in k8s context using `securityContext`. We need to run the container as the user `100` and group `101` as the container needs to access the headlamp directory.

For example in [values.yaml](https://github.com/headlamp-k8s/headlamp/blob/main/charts/headlamp/values.yaml#L66) we can just do.

```bash
securityContext:
  # capabilities:
  #   drop:
  #   - ALL
  # readOnlyRootFilesystem: true
  runAsNonRoot: true
  privileged: false
  runAsUser: 100
  runAsGroup: 101
```

It doesn't matter how you use init containers since they are build differently and does not have `headlamp` user implemented.